src/Compilers/Core/Portable/CodeGen/ILBuilderEmit.cs

```diff
@@ -109,7 +109,8 @@ internal void EmitStackAllocBlockInitializer(ImmutableArray<byte> data, SyntaxNo
             }
             else
             {
-                var field = module.GetFieldForData(data, syntaxNode, diagnostics);
+                // alignment==1 as emitInitBlock is false only if the element type for the stackalloc has SizeInBytes == 1.
```

Comment:
>// alignment==1 as emitInitBlock is false only if the element type for the stackalloc has SizeInBytes == 1. [](http://example.com/codeflow?start=16&length=107)

Rather than making this claim in a comment, consider inlining the code at the call site. It looks like there is only one call site per distinct ```emitInitBlock``` value. Then code will be explicitly guarded by the size check and we wouldn't need to worry about violating this assumption in the future,